Example/Sentinel/AppDelegate.swift

```diff
@@ -58,33 +58,117 @@ private extension AppDelegate {
         )
     }
-    
-    var optionSwitchItems: [OptionSwitchItem] {
-       [
-        OptionSwitchItem(
-            name: "Analytics",
-            setter: { AppSwitches.analyticsEnabled = $0 },
-            getter: { AppSwitches.analyticsEnabled },
-            userDefaults: .standard,
-            userDefaultsKey: "com.infinum.sentinel.optionSwitch.analytics"
-        ),
-        OptionSwitchItem(
-            name: "Crashlytics",
-            setter: { AppSwitches.crashlyticsEnabled = $0 },
-            getter: { AppSwitches.crashlyticsEnabled },
-            userDefaults: .standard,
-            userDefaultsKey: "com.infinum.sentinel.optionSwitch.crashlytics"
-        ),
-        OptionSwitchItem(
-            name: "Logging",
-            setter: { AppSwitches.loggingEnabled = $0 },
-            getter: { AppSwitches.loggingEnabled },
-            userDefaults: .standard,
-            userDefaultsKey: "com.infinum.sentinel.optionSwitch.logging"
-        ),
-       ]
 
+    /// Simple switch items that can be toggled on or off.
+    var preferences: [PreferenceItem] {
+        return [
+            // Old API values:
+            .init(
+                name: "Analytics",
+                setter: { AppSwitches.analyticsEnabled = $0 },
+                getter: { AppSwitches.analyticsEnabled },
+                userDefaults: .standard,
+                userDefaultsKey: "com.infinum.sentinel.optionSwitch.analytics"
+            ),
+            .init(
+                name: "Crashlytics",
+                setter: { AppSwitches.crashlyticsEnabled = $0 },
+                getter: { AppSwitches.crashlyticsEnabled },
+                userDefaults: .standard,
+                userDefaultsKey: "com.infinum.sentinel.optionSwitch.crashlytics"
+            ),
+            .init(
+                name: "Logging",
+                setter: { AppSwitches.loggingEnabled = $0 },
+                getter: { AppSwitches.loggingEnabled },
+                userDefaults: .standard,
+                userDefaultsKey: "com.infinum.sentinel.optionSwitch.logging"
+            ),
+            // New API values:
+            .init(
+                textName: "Name",
+                info: "From 1 to 5",
+                userDefaultsKey: "com.infinum.sentinel.optionSwitch.name",
+                onDidSet: nil
+            ),
+            .init(
+                integerName: "Star Rating",
+                info: "From 1 to 5",
+                min: 1,
+                max: 5,
+                userDefaultsKey: "com.infinum.sentinel.optionSwitch.starRating",
+                onDidSet: nil
+            ),
+            .init(
+                doubleName: "Temperature",
+                info: "Decimal number",
+                min: -50,
+                max: 50,
+                userDefaultsKey: "com.infinum.sentinel.optionSwitch.temperature",
+                onDidSet: { [weak self] in self?.didChange(temperature: $0) }
```

Comment:
You exposed what happens when the value changes, but preference items need both a getter and a setter.

The reason for the getter is an initial value or getting the value at a specific time. While the setter should be exposed to changing the value outside of this implementation, check how `Bool` preference values work.

We can discuss if an additional parameter `onDidSet` is needed because the value will be set automatically with a setter. This applies for all implementations in this file.

Reply:
For me it made more sense to simply use UserDefaults for both getting and setting the value. An optional `onDidSet` can be provided to apply the change right away.

If there should be a custom getter for this, then it should go another route and not use userDefaultsKey. I've already considered this and had an enum with either `userDefaults` or `custom` as cases of saving/retrieving the value, but it got too complex quickly.

Reply:
The problem is not user defaults but the suite you use. We should not assume that .standard is used. This can be different, based on use case (sharing between apps, testing suite, etc.).